ui/src/Pages/SearchBar.tsx

```diff
@@ -0,0 +1,166 @@
+  const [albums, setAlbums] = useState<Album[]>([]);
+  const [playlists, setPlaylists] = useState<Playlist[]>([]);
+  const [option, setOption] = useState("");
+  const accessToken = Auth();
+
+  const searchSpotify = () => {
+    const searchParameters = {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${accessToken}`,
+      },
+    };
+
+    fetch(
+      `https://api.spotify.com/v1/search?q=${searchInput}&type=artist,album,track,playlist`,
+      searchParameters
+    )
+      .then((response) => response.json())
+      .then((data: API) => {
+        console.log(data);
+        setArtists(data.artists.items);
+        setAlbums(data.albums.items);
+        setTracks(data.tracks.items);
+        setPlaylists(data.playlists.items);
+      });
+  };
+
+  const renderOption = () => {
+    if (option == "playlists") {
+      return (
+        <div className="bg-black mx-auto max-w-8xl py-5 sm:px-6 lg:px-8 ">
+          <div className="h-auto overflow-x-auto relative">
+            <h2 className="p-2 text-2xl font-bold text-white">Playlists</h2>
+            <Playlists playlist={playlists} n={20} />
+          </div>
+        </div>
+      );
+    } else if (option == "artists") {
+      return (
+        <div className="py-5 sm:px-6 lg:px-10 bg-black">
+          <h2 className="text-2xl font-bold text-white">Artists</h2>
+          <Artists artist={artists} n={20} />{" "}
+        </div>
+      );
+    } else if (option == "albums") {
+      return (
+        <div className="mx-auto max-w-8xl py-5 sm:px-6 lg:px-8 bg-black">
+          <h2 className="text-2xl font-bold text-white">Albums</h2>
+          <div className="inline-flex">
+            <Albums album={albums} n={20} />
+          </div>
+        </div>
+      );
+    } else if (option == "all") {
+      return (
+        <div>
+          <div className="py-5 sm:px-6 lg:px-10 bg-gray-100">
+            <h2 className="text-2xl font-bold text-zinc-900">Top Results</h2>
+            <div className="p-5">
+              <Artists artist={artists} n={5} />
+            </div>
+
+            <div className="h-screen overflow-x-auto mt-6 bg-gray-200 relative">
+              <Tracks track={tracks} albumImage="" />
+            </div>
+          </div>
+          <div className="bg-zinc-900 mx-auto max-w-8xl py-5 sm:px-6 lg:px-8 ">
+            <div className="h-auto overflow-x-auto relative">
+              <h2 className="p-2 text-2xl font-bold text-white">Playlists</h2>
+              <Playlists playlist={playlists} n={4} />
+            </div>
+          </div>
+          <div className="mx-auto max-w-8xl py-5 sm:px-6 lg:px-8 bg-black">
+            <h2 className="text-2xl font-bold text-white">Albums</h2>
+            <div className="inline-flex">
+              <Albums album={albums} n={10} />
+            </div>
+          </div>
+        </div>
+      );
+    }
+  };
```

Comment:
Not Appropriate way to render.
Especially using the option directly from state instead of parameter is not encouraged.
Try to create components out of them and shorted the code for better readability.